x/ccv/provider/keeper/power_shaping.go

```diff
@@ -551,3 +550,121 @@ func (k Keeper) DeleteMinimumPowerInTopN(
 	store := ctx.KVStore(k.storeKey)
 	store.Delete(types.MinimumPowerInTopNKey(consumerId))
 }
+
+// SetPrioritylist prioritylists validator with `providerAddr` address on chain `consumerId`
+func (k Keeper) SetPrioritylist(
+	ctx sdk.Context,
+	consumerId string,
+	providerAddr types.ProviderConsAddress,
+) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.PrioritylistKey(consumerId, providerAddr), []byte{})
+}
+
+// GetPriorityList returns all prioritylisted validators
+func (k Keeper) GetPriorityList(
+	ctx sdk.Context,
+	consumerId string,
+) (providerConsAddresses []types.ProviderConsAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.StringIdWithLenKey(types.PrioritylistKeyPrefix(), consumerId)
+	iterator := storetypes.KVStorePrefixIterator(store, key)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		providerConsAddresses = append(providerConsAddresses, types.NewProviderConsAddress(iterator.Key()[len(key):]))
+	}
+
+	return providerConsAddresses
+}
+
+// IsPrioritylisted returns `true` if validator with `providerAddr` has been prioritylisted on chain `consumerId`
+func (k Keeper) IsPrioritylisted(
+	ctx sdk.Context,
+	consumerId string,
+	providerAddr types.ProviderConsAddress,
+) bool {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.PrioritylistKey(consumerId, providerAddr))
```

Comment:
You can directly do `return store.Get(...)`

Reply:
I think it would then result in an error because it expects a boolean return value.

Reply:
You can do `return store.Get(...) != nil`, right?

Reply:
Yes, I'll do that. Thanks!